File: webui/src/lib/api/grok.ts
import type { GrokSessionDetail, GrokSessionSummary, GrokDeletePreview, GrokDeleteRequest, GrokDeleteResult } from "../../types";
import { callCommand, currentRuntimeContext } from "./transport";

export function listGrokSessions(query?: { q?: string; limit?: number }): Promise<GrokSessionSummary[]> {
  return callCommand<GrokSessionSummary[]>("grok.list", query);
}

export function renameGrokSession(id: string, title: string, csrfToken?: string | null): Promise<GrokSessionSummary> {
  return callCommand("grok.rename", { id, title, csrfToken });
}

export function previewGrokSessionDelete(id: string, csrfToken?: string | null): Promise<GrokDeletePreview> {
  return callCommand("grok.deletePreview", { id, csrfToken });
}

export function deleteGrokSession(request: GrokDeleteRequest, csrfToken?: string | null): Promise<GrokDeleteResult> {
  return callCommand("grok.deleteExecute", currentRuntimeContext().kind === "desktop" ? { request } : { ...request, csrfToken });
}

export function getGrokSession(id: string): Promise<GrokSessionDetail> {
  return callCommand<GrokSessionDetail>("grok.detail", { id });
}

File: src-tauri/src/commands/grok.rs
#![allow(non_snake_case)]

use nexushub_core::{
    grok::{GrokDeletePreview, GrokDeleteRequest, GrokDeleteResult},
    platform::PlatformPaths,
    services::use_cases::NexusHubUseCases,
};

#[tauri::command(rename = "grok.list")]
pub fn listGrokSessions(
    q: Option<String>,
    limit: Option<usize>,
) -> Result<Vec<nexushub_core::grok::GrokSessionSummary>, String> {
    NexusHubUseCases::new(&PlatformPaths::desktop_current())
        .grok()
        .list(limit.unwrap_or(100), q.as_deref())
        .map_err(|err| err.to_string())
}

#[tauri::command(rename = "grok.detail")]
pub fn getGrokSession(id: String) -> Result<nexushub_core::grok::GrokSessionDetail, String> {
    NexusHubUseCases::new(&PlatformPaths::desktop_current())
        .grok()
        .detail(&id)
        .map_err(|err| err.to_string())
}

#[tauri::command(rename = "grok.rename")]
pub async fn renameGrokSession(
    id: String,
    title: String,
) -> Result<nexushub_core::grok::GrokSessionSummary, String> {
    NexusHubUseCases::new(&PlatformPaths::desktop_current())
        .grok()
        .rename(&id, &title)
        .await
        .map_err(|err| err.to_string())
}

#[tauri::command(rename = "grok.deletePreview")]
pub fn previewGrokSessionDelete(id: String) -> Result<GrokDeletePreview, String> {
    NexusHubUseCases::new(&PlatformPaths::desktop_current())
        .grok()
        .delete_preview(&id)
        .map_err(|err| err.to_string())
}

#[tauri::command(rename = "grok.deleteExecute")]
pub fn deleteGrokSession(request: GrokDeleteRequest) -> Result<GrokDeleteResult, String> {
    NexusHubUseCases::new(&PlatformPaths::desktop_current())
        .grok()
        .delete_execute(request)
        .map_err(|err| err.to_string())
}
